detection_rules/rule_validators.py

```diff
@@ -93,68 +93,111 @@ def validate_stack_combos(self, data: QueryRuleData, meta: RuleMeta) -> Union[KQ
                 print(err_trailer)
                 return exc
 
-    def validate_integration(self, data: QueryRuleData, meta: RuleMeta, package_integrations: List[dict]) -> Union[
-            KQL_ERROR_TYPES, None, TypeError]:
+    def validate_integration(
+        self, data: QueryRuleData, meta: RuleMeta, package_integrations: List[dict]
+    ) -> Union[KQL_ERROR_TYPES, None, TypeError]:
         """Validate the query, called from the parent which contains [metadata] information."""
         if meta.query_schema_validation is False or meta.maturity == "deprecated":
-            # syntax only, which is done via self.ast
             return
 
         error_fields = {}
-        current_stack_version = ""
-        combined_schema = {}
-        for integration_schema_data in get_integration_schema_data(data, meta, package_integrations):
-            ecs_version = integration_schema_data['ecs_version']
-            integration = integration_schema_data['integration']
-            package = integration_schema_data['package']
-            package_version = integration_schema_data['package_version']
-            integration_schema = integration_schema_data['schema']
-            stack_version = integration_schema_data['stack_version']
-
-            if stack_version != current_stack_version:
-                # reset the combined schema for each stack version
-                current_stack_version = stack_version
-                combined_schema = {}
-
-            # add non-ecs-schema fields for edge cases not added to the integration
+        package_schemas = {}
+
+        # Initialize package_schemas with a nested structure
+        for integration_data in package_integrations:
+            package = integration_data["package"]
+            integration = integration_data["integration"]
+            if integration:
+                package_schemas.setdefault(package, {}).setdefault(integration, {})
+            else:
+                package_schemas.setdefault(package, {})
+
+        # Process each integration schema
+        for integration_schema_data in get_integration_schema_data(
+            data, meta, package_integrations
+        ):
+            package, integration = (
+                integration_schema_data["package"],
+                integration_schema_data["integration"],
+            )
+            integration_schema = integration_schema_data["schema"]
+
+            # Add non-ecs-schema fields
             for index_name in data.index:
                 integration_schema.update(**ecs.flatten(ecs.get_index_schema(index_name)))
 
-            # add endpoint schema fields for multi-line fields
+            # Add endpoint schema fields for multi-line fields
             integration_schema.update(**ecs.flatten(ecs.get_endpoint_schemas()))
-            combined_schema.update(**integration_schema)
+            if integration:
+                package_schemas[package][integration] = integration_schema
+            else:
```

Comment:
```suggestion
            elif package and not integration:
```

Reply:
what is the benefit of this suggestion?

Reply:
Not assuming package will always defined but relying on it in else logic.
Feel free to ignore.